core/geometry/src/geometry3d/FrameBuilder.ts

```diff
@@ -178,6 +179,14 @@ export class FrameBuilder {
             this.announcePoint(point);
           else break;
         }
+      } else if (data instanceof InterpolationCurve3d) {
+        const point = Point3d.create();
+        for (let i = 0; this.savedVectorCount() < 2; i++) {
+          if (i < data.options.fitPoints.length-1) {
```

Comment:
For a 3-pt curve, do you need i < data.options.fitPoints.length ?

Reply:
Don't think so, data.options.fitPoints.length isn't a valid point index. This seemed to return the right result, but I assume you will test/refactor when you implement something for a default/fallback. :)